core-processor/src/ext.rs

```diff
@@ -1154,17 +1157,24 @@ impl<LP: LazyPagesInterface> Externalities for Ext<LP> {
                     .gr_read_per_byte
                     .cost_for(len.into()),
             )?;
-            PayloadSliceLock::try_new((at, end), &mut mutator.ext.context.message_context)
-                .ok_or_else(|| FallibleExecutionError::ReadWrongRange.into())
-        })
-    }
 
-    fn unlock_payload(&mut self, payload_holder: &mut PayloadSliceLock) -> UnlockPayloadBound {
-        UnlockPayloadBound::from((&mut self.context.message_context, payload_holder))
+            PayloadSlice::try_new(
+                at,
```

Comment:
in this place we do have invariant for end >= at, but it'd be nice to have also inside PayloadSlice::try_new as well just in case

Reply:
Added this check